### ISSUE TITLE
QIO + Tunecache problem

### ISSUE THREAD

**Issue by cpviolator** [lattice/quda#572]
Recent commits to develop place the bulk of the saveVector(B) function inside a `HAVE_QIO` preprocessor flag. Using multigrid with these commits, and not using QIO, results in a call to errorQuda.

I placed a guard around the call to saveVectors(B):
```c++
if (strcmp(param.mg_global.vec_outfile,"")!=0) { // only save if outfile is defined                                                                       
      saveVectors(B);
    }
```
but then I consistently get failures from the tuning routines during `Computing TMCAV` (or `Computing AV` for wilson/clover):
```c++
MG level 1 (GPU): ERROR: Error writing name string (rank 3, host nid03827, /scratch/snx3000/howarth/QUDA-GIT-TEAM/quda-develop/quda/lib/tune.cpp:80 in deserializeTuneCache())
MG level 1 (GPU):        last kernel called was (name=N4quda10CalculateYILb0EfLi4ELi3ELi2ELi24ENS_13CalculateYArgIfNS_5gauge10FieldOrderIfLi48ELi2EL21QudaGaugeFieldOrder_s5EEENS3_IfLi3ELi1ELS4_5EEENS_11colorspinor12FieldOrderCBIfLi4ELi3ELi24EL16QudaFieldOrder_s5EEESA_NS_6clover10FieldOrderIfLi3ELi4EL22QudaClovol=165888,stride=82944,precision=4,TwistFlavour=1,comm=0001,computeTmcAV,CPU,volume=24x24x24x12,aux=vol=165888,stride=82944,precision=4,TwistFlavour=1,comm=0001,computeTmcAV,CPU)
``` 
Is this a known issue?

CUDA 8.0
Cray CC 2.5.8
CMake 3.6.2
QUDA 0.9.0 (git v0.8.0-1572-g2dd7948-dirty-sm_60)

**Comment by maddyscientist** [comment]
I noticed this issue earlier today.  Fix is trivial
```
--- a/include/tune_key.h
+++ b/include/tune_key.h
@@ -8,7 +8,7 @@ namespace quda {
   struct TuneKey {
 
     static const int volume_n = 32;
-    static const int name_n = 256;
+    static const int name_n = 384;
     static const int aux_n = 256;
     char volume[volume_n];
     char name[name_n];
```
I'll add this fix to the BQCD pull right now so it will be picked up tomorrow and merged into develop.

**Comment by maddyscientist** [comment]
Pushed a fix to hotfix/bqcd-copy with 5d3831a6a8cbbc05471dbc474ceba6214b3c8173.

**Comment by cpviolator** [comment]
Excellent. I was trawling the history of tune.cpp and saw this change in header. Thanks for the quick feedback.

BTW, can we also implement the guard around saveVectors(B) to prevent the call to errorQuda if QIO is not installed?

**Comment by maddyscientist** [comment]
Sure, we add that guard.  Add it to your current pull if you want, and I will see we can merge the pull this week.  Just getting back into the multigrid stuff now (in preparation for hackathon next week) so cleaning up and testing my main development branch (feature/multigrid).  Lots of improvements there that are coming to a GPU near you real soon. 😄

**Comment by cpviolator** [comment]
Fixed with https://github.com/lattice/quda/commit/5d3831a6a8cbbc05471dbc474ceba6214b3c8173

**Comment by maddyscientist** [comment]
Not that it matters, but if you state in you commit comment that this commit closes a given issue, e.g.,
```
git commit -m"Closes #572"
```
then when the branch that the commit is in is merged into the develop branch github will automatically close the issue.

**Comment by cpviolator** [comment]
Great! I love learning new git magic, thanks Kate.
